### PR TITLE
Prevent printing when using ContinueOnError

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

File: flag.go [hunk 1]
```diff
@@ -1056,6 +1058,7 @@ func (f *FlagSet) Parse(arguments []string) error {
 		case ContinueOnError:
 			return err
 		case ExitOnError:
+			fmt.Println(err)
```
Comment (bogem) [inline flag.go]: Actually, it's a bad idea to integrate it. User should handle error by himself/herself (use log or fmt). And with this spf13/cobra prints error twice. Please delete it.

Comment (stevenroose) [reply flag.go]: How can Cobra print after os.Exit is called? I agree that the user should be able to decide, but I assumed that after os.Exit had been called, there was no way to show the error anymore.

Comment (stevenroose) [reply flag.go]: > Exit causes the current program to exit with the given status code. Conventionally, code zero indicates success, non-zero an error. The program terminates immediately; deferred functions are not run. 

So I believe without that line, all prints would be gone. I think Cobra would use ContinueOnError, so this PR would not make it print anything.

Comment (bogem) [reply flag.go]: Oh, it was my dumb fault 😓
You're right.
Anyway cobra uses ContinueOnError as errorHandling.

**Comment by cornfeedhobo** [comment]
@stevenroose I know it's been a long time, but do you know the reason you made this change? It's interesting because it deviates from what golang's flag package does.

**Comment by stevenroose** [comment]
> @stevenroose I know it's been a long time, but do you know the reason you made this change? It's interesting because it deviates from what golang's flag package does, and when reverted, cobra and pflag tests pass.

@cornfeedhobo I don't remember. I remember trying to get cobra etc to do what I wanted, but eventually giving up and creating [gonfig](https://github.com/stevenroose/gonfig). But I have not been using Go for almost 3 years now.

**Comment by cornfeedhobo** [comment]
@stevenroose I know how it goes. Thanks for the prompt reply!

**Pull request merged**